File: libs/language/src/repo/core/schema/TimelineChannel.ts
import { z } from 'zod';
import { blueIds } from '../blue-ids';
import { withTypeBlueId } from '../../../schema/annotations/typeBlueId';
import { ChannelSchema } from './Channel';

export const TimelineChannelSchema = withTypeBlueId(
  blueIds['Timeline Channel']
)(
  ChannelSchema.extend({
    timelineId: z.string().optional(),
  })
);

export type TimelineChannel = z.infer<typeof TimelineChannelSchema>;

File: references/java-sdk/src/main/java/blue/language/types/conversation/TimelineChannel.java
package blue.language.types.conversation;

import blue.language.model.TypeBlueId;
import blue.language.model.Node;
import blue.language.types.TypeAlias;

@TypeAlias("Conversation/Timeline Channel")
@TypeBlueId("EvuCWsG1E6WJQg8QXmk6rwMANYTZjoLWVZ1vYQWUwdTH")
public class TimelineChannel {
    public String timelineId;
    public Node event;

    public TimelineChannel timelineId(String timelineId) {
        this.timelineId = timelineId;
        return this;
    }

    public TimelineChannel event(Node event) {
        this.event = event;
        return this;
    }
}
